core/src/main/java/io/grpc/AbstractServerBuilder.java

```diff
@@ -114,18 +111,39 @@ public ServerImpl build() {
       releaseExecutor = true;
     }
 
-    ServerImpl server = new ServerImpl(executor, registry);
-    server.setTransportServer(buildTransportServer(server.serverListener()));
+    final ServerEssentials essentials = buildEssentials();
+    ServerImpl server = new ServerImpl(executor, registry, essentials.server);
     server.setTerminationRunnable(new Runnable() {
       @Override
       public void run() {
         if (releaseExecutor) {
           SharedResourceHolder.release(DEFAULT_EXECUTOR, executor);
         }
+        if (essentials.terminationRunnable != null) {
```

Comment:
Hmm... Maybe, but it seems like it doesn't matter. I was mimicking AbstractChannelBuilder. If this needs to be changed I'll update AbstractChannelBuilder as well.